quinn-proto/src/crypto.rs

```diff
@@ -101,6 +125,11 @@ impl CryptoSession for TlsSession {
 }
 
 pub trait CryptoSession {
+    fn alert(&self) -> Option<AlertDescription>;
```

Comment:
We could replace `AlertDescription` with `u8` here and elsewhere, if desired. I only ever included it for the sake of `TransportError`'s `Debug` impl, but that re-parses it from a byte anyway.

Reply:
I'd like to leave it for now -- this effort isn't done yet anyway and this is clearly a TLS detail seeping through, so it will need changes one way or another down the line.

Reply:
Could still be useful for swapping in other TLS impls if someone wants to do that, but we can worry about it when such a requirement actually arises.

Reply:
My goal is to make this API abstract enough that it can cover non-TLS crypto protocols like Noise, too, so I'd prefer not to expose this TLSicism.

Reply:
It's worth noting that QUIC reserves exactly one `u8` worth of transport error codes "for carrying error codes specific to the cryptographic handshake that is used", so while `fn alert(&self)` might not be the precise abstraction we want, whatever we do land on should probably have `u8` error codes all the same.